### ISSUE TITLE
Annotation with multiple classes

### ISSUE THREAD

**Issue by Woyahdrem** [matterport/Mask_RCNN#2141]
Hi

I am trying to train a model to recognize what color a certain type of object contains.

My problem is that any these objects may contain zero or more colors, and while right now I am limiting the search on only two colors, in the future this number may increase.
And since the number of possible combinations increase exponentially with the number of colors it seems to me that having one class for each of those combinations is not the best approach.

Ideally, I would use one class for each color (or two, for example "red" and "not_red" and one of those must be selected), but this would mean that:
1. The model must be able to assign more to one class to each pixel, and potentially the number of simultaneous classes is different for each pixel.
2. The variable class_ids is no longer a 1D array (each mask has one class), but a 2D array (each mask can have more than one class).

While I have no idea if [1] would be a problem, I'm pretty sure [2] would be a problem since the code assumes that the variable class_ids returned by Dataset.load_mask() is a 1D array.

I also thought about training multiple models, one for each class, and then somehow merge their outputs, but I am not sure on how I should proceed in this case.

What would be my best option?
Is this even possible, or should I just use one class for each combination of colors while excluding the least likely combinations?

**Comment by susnato** [comment]
I don't think this is possible to train a model to recognize what colour a certain type of object contains. Since a  creator or a designer can use any colour they want ....but in some cases, it's possible like car tyres are always black, grass and trees are always green but you can't use the model in a big scale like in cars are other objects.
But I could be wrong.

**Comment by Woyahdrem** [comment]
My objective is actually much simpler.

The object to recognize is always the same: a circle (or circular-ish shape).
Any circle may contain colored dots, and I have to classify the circles depending on the dots contained.

My true problem is effectively the high combination of colored dots (or, from another point of view, the high number of different types of circles) that needs to be identified.

I could simply have any circle containing a combination of colors be defined as a class, but the number of classes would be equal to 2 to the power of the numbers of colors. 
And since in the future I plan on expanding on a dozen of colors, that would lead to having thousands of classes.

**Comment by susnato** [comment]
I think you should write two algorithms the first one detects the circle(I mean creates a mask on the circle) and the next one detects the types of dots in the circle. By this, you don't have to deal with the exponential growth of classes. 
If you need any more answers you can mail me. : )

**Comment by truongtd6285** [comment]
I think you can use mask RCNN to detect all objects as 1 class, as they have the same shape. Then you can use a tool like Opencv to extract the object and calculate the number of color inside each object.
